src/lightning_app/components/database/client.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Dict, List, Optional, Type, TypeVar
+
+import requests
+from requests import Session
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+from lightning_app.components.database.utilities import GeneralModel
+
+_CONNECTION_RETRY_TOTAL = 5
+_CONNECTION_RETRY_BACKOFF_FACTOR = 1
+
+
+def _configure_session() -> Session:
+    """Configures the session for GET and POST requests.
+
+    It enables a generous retrial strategy that waits for the application server to connect.
+    """
+    retry_strategy = Retry(
+        # wait time between retries increases exponentially according to: backoff_factor * (2 ** (retry - 1))
+        total=_CONNECTION_RETRY_TOTAL,
+        backoff_factor=_CONNECTION_RETRY_BACKOFF_FACTOR,
+        status_forcelist=[429, 500, 502, 503, 504],
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+    http = requests.Session()
+    http.mount("https://", adapter)
+    http.mount("http://", adapter)
+    return http
+
+
+T = TypeVar("T")
+
+
+class DatabaseClient:
+    def __init__(self, db_url: str, token: str, model: Optional[T] = None) -> None:
+        self.db_url = db_url
+        self.model = model
+        self.token = token
+        self._session = None
+
+    def select_all(self, model: Optional[Type[T]] = None) -> List[T]:
+        cls = model if model else self.model
+        assert cls
```

Comment:
are these asserts  necessary? Generally asserts within the code should be avoided if possible

Reply:
Yes, better if no model has been provided in either init nor select.